CONTRIBUTING.md

```diff
@@ -0,0 +1,66 @@
+#  Contributing
+
+From opening a bug report to creating a pull request: we appreciate and welcome
+every contribution. If you're planning to implement a new feature or change
+the API please create an issue first. This way we can ensure that your precious
+work is not in vain.
+
+There is a series of blog posts over at the [website](https://martyav.github.io/), detailing the experiences of Marty Hernandez Avedon at Wikimedia. This one in particular should be helpful for people working on the [documentation project](https://martyav.github.io/2019-01-03-wikimedia-week-5/)
+
+### Installation
+```
+$ git clone https://github.com/wikimedia/MediaWiki-Action-API-Code-Samples.git
+$ cd MediaWiki-Action-API-Code-Samples
+Install the necessary python modules with pip
+$ python3 name_of_the_file.py #Enter any credentials if required in the file
+```
+
+### Contributing code samples
+First, propose an idea for a code sample, demo app, etc. by creating an issue around it in the repository. After discussing your idea with the repo contributors, start working, and then send a pull request, when you've your changes ready to be accepted/ merged! You can autogenerate python files for GET Requests demos where feasible by following the instructions below:
+```
+$ cd MediaWiki-Action-API-Code-Samples
+$ Add module information to `modules.json`
+$ cd python
+$ python autogenerator.py
+$ Make desired changes to the newly generated file(s)
+```
+
+## Issues
+
+If you are having difficulty after looking over your configuration carefully, please post
+a question to [StackOverflow with the mediawiki tag](http://stackoverflow.com/tags/mediawiki).
+
+**If you have discovered a bug or have a feature suggestion, please [create an issue on GitHub](https://github.com/wikimedia/MediaWiki-Action-API-Code-Samples/issues/new).**
+
+Do you want to fix an issue?  Look at the [issues](https://github.com/wikimedia/MediaWiki-Action-API-Code-Samples/issues).
```

Comment:
Just a nitpick but maybe:
 
_Want to go ahead and fix some bugs or add features? Take a look at the open [issues]_